src/Uno.UI/UI/Xaml/FrameworkElement.cs

```diff
@@ -357,6 +365,90 @@ private void UpdateActiveStyle()
 
 		private Style ResolveImplicitStyle() => (this as IDependencyObjectStoreProvider).Store.GetImplicitStyle();
 
+		public ElementTheme RequestedTheme
+		{
+			get => (ElementTheme)GetValue(RequestedThemeProperty);
+			set => SetValue(RequestedThemeProperty, value);
+		}
+
+		public static DependencyProperty RequestedThemeProperty { get; } =
+			DependencyProperty.Register(
+				nameof(RequestedTheme), typeof(ElementTheme),
+				typeof(FrameworkElement),
+				new FrameworkPropertyMetadata(ElementTheme.Default, FrameworkPropertyMetadataOptions.Inherits, RequestedThemePropertyChanged));
+
+		private static void RequestedThemePropertyChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+		{
+			var frameworkElement = (FrameworkElement)dependencyObject;
+
+			void PropagateThemeChanged(object instance)
+			{
+				// Update ThemeResource references that have changed
+				if (instance is FrameworkElement fe)
+				{
+					fe.UpdateThemeBindings();
+					fe.UpdateActualTheme();
+				}
+
+				//Try Panel.Children before ViewGroup.GetChildren - this results in fewer allocations
+				if (instance is Controls.Panel p)
+				{
+					foreach (object o in p.Children)
+					{
+						PropagateThemeChanged(o);
+					}
+				}
+				else if (instance is ViewGroup g)
+				{
+					foreach (object o in g.GetChildren())
+					{
+						PropagateThemeChanged(o);
+					}
+				}
+			}
+			PropagateThemeChanged(frameworkElement);			
+		}		
+
+		public ElementTheme ActualTheme
+		{
+			get => (ElementTheme)GetValue(ActualThemeProperty);
+		}
+
+		public static DependencyProperty ActualThemeProperty { get; } =
+			DependencyProperty.Register(
+				nameof(ActualTheme), typeof(ElementTheme),
+				typeof(FrameworkElement),
+				new PropertyMetadata(ElementTheme.Default, ActualThemePropertyChanged));
```

Comment:
Should this property be property inherited ?

Reply:
Adjusted :-)